### PR TITLE
Added warning for 32bit Python

### PR DESCRIPTION
**Pull request by haydensheung**
Please see my code!

### CODE REVIEW

File: client/cli/ok.py [hunk 1]
```diff
@@ -92,6 +93,12 @@ def parse_input(command_input=None):
 
 def main():
     """Run all relevant aspects of ok.py."""
+
+    #Checking user's Python bit version
```
Comment (albert12132) [inline client/cli/ok.py]: Please move this block below `log.debug(args)`. All log messages should be made after we set the log level.

File: client/cli/ok.py [hunk 2]
```diff
@@ -92,6 +93,12 @@ def parse_input(command_input=None):
...
+    bit_v = (8 * struct.calcsize("P"))
+    if (bit_v == 32):
```
Comment (albert12132) [inline client/cli/ok.py]: Style nit: parentheses are not required for if statements in Python.

File: client/cli/ok.py [hunk 3]
```diff
@@ -92,6 +93,12 @@ def parse_input(command_input=None):
...
+        print("32 bit Python is not supported. Please install the 64 bit version.")
+        log.debug("32 bit Python is detected.")
```
Comment (albert12132) [inline client/cli/ok.py]: Doing both logging and printing is kind of redundant here. Since we definitely want to print this message, you can omit the log message.

File: client/cli/ok.py [hunk 4]
```diff
@@ -92,6 +93,12 @@ def parse_input(command_input=None):
 
 def main():
     """Run all relevant aspects of ok.py."""
+
+    #Checking user's Python bit version
+    bit_v = (8 * struct.calcsize("P"))
+    if (bit_v == 32):
+        print("32 bit Python is not supported. Please install the 64 bit version.")
```
Comment (albert12132) [inline client/cli/ok.py]: It might be strange to a student if they see this message, but OK works anyway ("OK worked this time. Do I really need to update to 64 bit?").

Another concern is that a typical OK session prints a lot of output to the screen, so this message will probably be lost and rarely seen by the the student. I suggest exiting immediately if 32-bit Python is detected (`exit(1)`). I'm not familiar with the 32-bit issue though; @Sumukh , what do you think?

File: client/cli/ok.py (hunk 3 above)
Comment (haydensheung) [inline client/cli/ok.py]: I believe @Sumukh wants me to log it to debug?

File: client/cli/ok.py (hunk 4 above)
Comment (Sumukh) [inline client/cli/ok.py]: This is especially concerning when 32bit errors with SSL result in tracebacks (see #152). The traceback itself will consume the screen. I don't think we should automatically give up with `exit(0)` just because there is an auth error. 

I propose the following: 
- Always log whether the client is 32bit or 64bit 
- Except the SSL errors and auth errors - if those error are raised while running the protocols, except them and ensure the bit version is 64 bit. If it's not - print a string about installing 64 bit

This effectively combines #152 and this PR. If you're ok with that course of action, Hayden's PR should just be limited to logging the bit version. 

This would mean that we'd still allow students to run it locally and have software updates etc.

Comment (albert12132) [inline client/cli/ok.py]: Sounds good to me.

Comment (haydensheung) [inline client/cli/ok.py]: So I change anything to this PR before merging then?

Comment (albert12132) [inline client/cli/ok.py]: Following our discussion, looks like this PR should just log the bit version regardless of whether or not it's 32 bit, and not print a message to the user. So effectively:

```python
log.setLevel(...)
bit_v = 8 * struct.calcsize("P")
log.debug("Python bit version: {}".format(bit_v))
```

@Sumukh do you intend for the second step of your proposal to be done in this PR?

Comment (Sumukh) [inline client/cli/ok.py]: Let'd do the second part in another PR/branch.

Comment (haydensheung) [inline client/cli/ok.py]: Okay, sounds good! Let me do that step right now.

Comment (haydensheung) [inline client/cli/ok.py]: I thought I added a comment here but apparently Github didn't get it -- what does `setlevel()` does?

Comment (albert12132) [inline client/cli/ok.py]: log.setlevel tells the logger to only emit log messages that are at or
above a certain level. We already set the log level in main. I just
included it in my comment to show that you should put your code under the
line that sets the log level.

Comment (haydensheung) [inline client/cli/ok.py]: I see, thanks!

**Comment by haydensheung** [comment]
Update based on Albert's comment!

File: client/cli/ok.py [hunk 5]
```diff
@@ -92,9 +93,15 @@ def parse_input(command_input=None):
 
 def main():
     """Run all relevant aspects of ok.py."""
+
```
Comment (albert12132) [inline client/cli/ok.py]: Please delete this blank line.

File: client/cli/ok.py [hunk 6]
```diff
@@ -92,9 +93,15 @@ def parse_input(command_input=None):
...
     args = parse_input()
 
     log.setLevel(logging.DEBUG if args.debug else logging.ERROR)
+    
+    #Checking user's Python bit version
```
Comment (albert12132) [inline client/cli/ok.py]: Please add a space after the `#` and add a period at the end.

**Comment by albert12132** [comment]
Just a few nits. Other than that, LGTM. Feel free to merge when you've resolved the comments.
